### DISCUSSION TITLE
Replace variables in doc

### DISCUSSION THREAD

**Discussion by barryvdh** [gotenberg/gotenberg#405] [Q&A]
Is it possible to use docx templates, to replace variables/fill a doc? Similar to https://github.com/open-xml-templating/docxtemplater
Or would it be possible to add that binary to Gotenberg? Eg. To load a template, fill it with invoice lines and then convert to PDF.

**Comment by gulien** [comment]
Hello @barryvdh,

Although I like the idea, I think it's a bit out-of-scope for Gotenberg. Right now, your best option is to create a dedicated service, for instance: https://github.com/EC-Nordbund/docx-templates-to-pdf.

You may also switch to HTML 😄

**Reply by stefangweichinger** [reply]
@barryvdh did you succeed? How did you solve this?

**Comment by MadhabaPatra** [comment]
Any solutions on this?

**Reply by barryvdh** [reply]
We created a docker with an endpoint that converts it with docxtemplater first and then to gotenberg

**Reply by MadhabaPatra** [reply]
> We created a docker with an endpoint that converts it with docxtemplater first and then to gotenberg

Great. Do you mind to share the docker? It'll be helpful for many of us.
